Week3/homework/2-step3.js

```diff
@@ -3,6 +3,31 @@
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const values = [];
   // Add your code here
+  for (let i = startIndex; i <= stopIndex; i++) {
+
+    values.push(i);
+
+    if (i % 3 === 0 && i % 5 === 0) {  // here the order of checking is important.
+      sayThree();
+      sayFive();
+    } else if (i % 3 === 0) {
+      sayThree();
+    } else if (i % 5 === 0) {
+      sayFive();
+    }
+
+  }
+  console.log(values); // just to print out the array when the programme run
```

Comment:
This is nice. Most people have been creating the values array first, and then processing the indexes (I also did this) but your way means you only traverse the whole structure once and do all your processing in one go.  Nice.